components/Typography/src/MDCFontScaler.h

```diff
@@ -0,0 +1,60 @@
+#import <UIKit/UIKit.h>
+
+// Material's text styles, which are similar, but not quite equivalent to Apple's UIFontTextStyle.
+extern NSString * _Nonnull MaterialTextStyleHeadline1;
+extern NSString * _Nonnull MaterialTextStyleHeadline2;
+extern NSString * _Nonnull MaterialTextStyleHeadline3;
+extern NSString * _Nonnull MaterialTextStyleHeadline4;
+extern NSString * _Nonnull MaterialTextStyleHeadline5;
+extern NSString * _Nonnull MaterialTextStyleHeadline6;
+extern NSString * _Nonnull MaterialTextStyleSubtitle1;
+extern NSString * _Nonnull MaterialTextStyleSubtitle2;
+extern NSString * _Nonnull MaterialTextStyleBody1;
+extern NSString * _Nonnull MaterialTextStyleBody2;
+extern NSString * _Nonnull MaterialTextStyleButton;
+extern NSString * _Nonnull MaterialTextStyleCaption;
+extern NSString * _Nonnull MaterialTextStyleOverline;
+
+
+/**
+ MDCFontScaler is used to attach a scaling curve to a UIFont via an associated object on that
+ font instance.
+
+ This interface is similar to UIFontMetrics, but the font returned from MDCFontScaler do *not*
+ automatically adjust when the device's text size / content size category is changed.
+ */
+@interface MDCFontScaler : NSObject
```

Comment:
We will need to allow subclassing to add the Internal-specific curves.

Reply:
I'm going to resolve this, but in case I wasn't clear, let me add a bit more context.
Clients may have their own implementation of type scaling or hierarchies they want to use.  So PanAmMaterial may have a FlightInfo text style (MDCTextStyle).  We need to allow PanAm to subclass to implement their own curves.  The common case would be to just call the existing MDC implementation (return super.typeScaleForTextStyle:) unless text style is unique to PanAm.